listenbrainz/db/msid_mbid_mapping.py

```diff
@@ -79,47 +80,49 @@ def fetch_track_metadata_for_items(items: List[ModelT]) -> List[ModelT]:
+    # same recording twice. since the dict is keyed by mbid/msid the corresponding value
+    # should be a iterable of all items having that mbid
+    msid_item_map, mbid_item_map = defaultdict(list), defaultdict(list)
     for item in items:
         if item.recording_mbid:
-            mbid_item_map[item.recording_mbid] = item
+            mbid_item_map[item.recording_mbid].append(item)
         else:
-            msid_item_map[item.recording_msid] = item
+            msid_item_map[item.recording_msid].append(item)
 
     msid_metadatas = load_recordings_from_msids(msid_item_map.keys())
     for metadata in msid_metadatas:
         msid = metadata["ids"]["recording_msid"]
-        item = msid_item_map[msid]
-        item.track_metadata = {
-            "track_name": metadata["payload"]["title"],
-            "artist_name": metadata["payload"]["artist"],
-            "additional_info": {
-                "recording_msid": msid
+        for item in msid_item_map[msid]:
+            item.track_metadata = {
+                "track_name": metadata["payload"]["title"],
+                "artist_name": metadata["payload"]["artist"],
+                "additional_info": {
+                    "recording_msid": msid
+                }
             }
-        }
 
     mapping_mbid_metadata, mapping_msid_metadata = load_recordings_from_mapping(mbid_item_map.keys(), msid_item_map.keys())
     _update_items_from_map(mbid_item_map, mapping_mbid_metadata)
     _update_items_from_map(msid_item_map, mapping_msid_metadata)
     return items
 
 
-def _update_items_from_map(models: Dict[str, ModelT], metadatas: Dict) -> Dict[str, ModelT]:
-    for _id, item in models.items():
+def _update_items_from_map(models: Dict[str, Iterable[ModelT]], metadatas: Dict):
+    for _id, items in models.items():
         if _id not in metadatas:
             continue
 
-        metadata = metadatas[_id]
-        item.track_metadata = {
-            "track_name": metadata["title"],
-            "artist_name": metadata["artist"],
-            "release_name": metadata["release"],
-            "additional_info": {
-                "recording_msid": metadata["recording_msid"],
-                "recording_mbid": metadata["recording_mbid"],
-                "release_mbid": metadata["release_mbid"],
-                "artist_mbids": metadata["artist_mbids"]
+        for item in items:
+            metadata = metadatas[_id]
+            item.track_metadata = {
+                "track_name": metadata["title"],
+                "artist_name": metadata["artist"],
+                "release_name": metadata["release"],
+                "additional_info": {
+                    "recording_msid": metadata["recording_msid"],
+                    "recording_mbid": metadata["recording_mbid"],
+                    "release_mbid": metadata["release_mbid"],
+                    "artist_mbids": metadata["artist_mbids"]
```

Comment:
not directly related to this PR, but are we always going to be sure that the `metadata` dict has all of these fields?

Reply:
Yes, these are the exact fields selected in `load_recordings_from_mapping` (above in this file) and this function is only called with data from that query.